server/api/jurisdictions.py

```diff
@@ -804,9 +814,110 @@ def get_batch_comparison_audit_discrepancies_by_jurisdiction(
+    # but some sampled ballots have ImprintedIDs of a different format that are
+    # very long and unreadable, so here we ensure they all have the same format
+    # Ex. "Tabulator X, Batch Y, Ballot Position Z, Container 0"
+    sampled_ballot_id_to_readable_identifier = dict(
+        (
+            sampled_ballot_id,
+            f"{tabulator}, {name}, Ballot Position {ballot_position}"
+            + (f", Container {container}" if container is not None else ""),
+        )
+        for sampled_ballot_id, tabulator, name, ballot_position, container in SampledBallot.query.filter(
+            SampledBallot.id.in_(ballots_in_round)
+        )
+        .join(Batch)
+        .with_entities(
+            SampledBallot.id,
+            Batch.tabulator,
+            Batch.name,
+            SampledBallot.ballot_position,
+            Batch.container,
+        )
+    )
+
+    for contest in election.contests:
+        audited_results = sampled_ballot_interpretations_to_cvrs(contest)
+        reported_results = cvrs_for_contest(contest)
+        for ballot_id, audited_result in audited_results.items():
+            audited_cvr = audited_result["cvr"]
+            reported_cvr = reported_results.get(ballot_id)
+            vote_deltas = ballot_vote_deltas(contest, reported_cvr, audited_cvr)
+            if not vote_deltas or isinstance(vote_deltas, str):
+                continue
+
+            reported_votes = {}
+            if reported_cvr and contest.id in reported_cvr:
+                reported_votes = reported_cvr[contest.id]
+
+            audited_votes = {}
+            if audited_cvr and contest.id in audited_cvr:
+                audited_votes = audited_cvr[contest.id]
+
+            if ballot_id in sampled_ballot_id_to_jurisdiction_id:
+                if not reported_cvr:
+                    reported_cvr = {}
+                jurisdiction_id = sampled_ballot_id_to_jurisdiction_id[ballot_id]
+                readable_ballot_identifier = sampled_ballot_id_to_readable_identifier[
+                    ballot_id
+                ]
+                discrepancies_by_jurisdiction[jurisdiction_id][
+                    readable_ballot_identifier
+                ][contest.id] = {
+                    "reportedVotes": normalize_vals_for_discrepancies(reported_votes),
+                    "auditedVotes": normalize_vals_for_discrepancies(audited_votes),
+                    "discrepancies": normalize_vals_for_discrepancies(vote_deltas),
+                }
+
+    return discrepancies_by_jurisdiction
+
+
+# normalize_vals_for_discrepancies normalizes the return type for batch and ballot comparison discrepanices.
+# It converts string numbers to int numbers, leaving non-numbers as is (i.e. 'o', 'u'). This is useful for
+# ballot CVRs to ensure that numbers are always returned as numbers, to match the behavior of batch discrepancies
+# and of vote deltas
+def normalize_vals_for_discrepancies(
```

Comment:
Ballot comparison cvrs are different than their batch comparison equivalent in that the values are all strings vs ints, to support 'o' and 'u'. I thought it would be better to be consistent in returning ints where it is a number, so that is the purpose of this function. It allows us to have a single return type for discrepancies that is relatively simple. Feel free to let me know if it doesn't seem right

Reply:
What happens if we just leave them as their internal representation (ints for batch comparison, strings for ballot comparison)? Might be simpler to just keep the same representation until there's somewhere we actually need to use the strings as ints, that way you don't need to think about whether it's an int in one place or a string in other places

Reply:
Hm that's true. I guess I thought it might be a bit odd if the frontend is getting strings on some API calls and ints on others for the same field values, when they might both be representing a 1, so that is what motivated me to normalize it. But, leaving them as is will work for sure too.

Reply:
Removed normalization [here](https://github.com/votingworks/arlo/pull/2034/commits/808cfeb3afd121a738256ffd1682da7e6430f754)